tests/test_scrape_github_trending.py

```diff
@@ -102,3 +102,88 @@ def test_fake_response_raises_on_error():
     response = FakeResponse(status_code=500)
     with pytest.raises(Exception):
         response.raise_for_status()
+
+
+def test_scrape_github_trending_skips_duplicates_precheck(monkeypatch):
+    import scrape_github_trending
+
+    html = """
+    <article class="Box-row">
+        <h2 class="h3"><a href="/owner/dupe">Dupe Tool</a></h2>
+        <p>Duplicate candidate</p>
+    </article>
+    """
+    response = FakeResponse(content=html.encode("utf-8"))
+    monkeypatch.setattr("scrape_github_trending.requests.get", lambda *args, **kwargs: response)
+
+    # Classifier marks it as a devtool
+    def fake_classify(candidates):
+        candidates = list(candidates)
+        return {item["id"]: True for item in candidates}
+
+    monkeypatch.setattr("scrape_github_trending.classify_candidates", fake_classify)
+    monkeypatch.setattr("scrape_github_trending.get_devtools_category", lambda *args, **kwargs: None)
+
+    # is_duplicate pre-check forces skip
```

Comment:
Since we stub `is_duplicate` here, we never exercise the real SQLite duplicate guard. Could we add an integration-style test with a temp DB to verify duplicates are skipped without calling `save_startup`, so DB-level regressions are caught?